src/libraries/System.Private.CoreLib/src/System/Number.Parsing.cs

```diff
@@ -1951,13 +1951,22 @@ internal static bool TryStringToNumber(ReadOnlySpan<char> value, NumberStyles st
 
         private static bool TrailingZeros(ReadOnlySpan<char> value, int index)
         {
-            return TrailingZeros(value.Slice(index));
+            // For compatibility, we need to allow trailing zeros at the end of a number string
+            for (int i = index; (uint)i < (uint)value.Length; i++)
```

Comment:
Why does this need to be expanded? I'd expect the JIT to do a good job with the inline here

Reply:
It doesn't inline it by default, I can force it, but in any case the code is bigger. It is a small improvement, but I'm trying to figure out why some micro-benchmarks are slower : [SharpLab](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIAYACY8gOgCUBXAOwwEt8YLAMIR8AB14AbGFADKMgG68wMXAG4aNYgGYmpBkIYBvGgzPmzYqLwXYMMJuSQNgECJIYAVKNim8uAOYAWjIQuAAUbDDYACYA8lySAJ6yYthcADxgABbYUAB8DLaSHDBoDP4YFVwxMAgAlKYW5ibUze0MAPSdDABi0AyQ4na8wH4YSeUA7g5cMDAxDBgQDNiSkhBTSz5+gQwAXqG4q1UY2Q4wNQwQAGarDFwc+MAyjnRNHeY3A+GVFQwAXmqtQQagY4Q4lXqvAYGXBkJ49WKpRYABlLgEzmDeABqHGNNqfFofIkWXh3cLImAAbV4AF0GABCIEAcgAOnQWQTSe1WjyicQAOwMG5rXAwDSE/lmAC+JM+cuo8o6Qu2pUlCpJyss1ls9kczlc7i8O0k/mCR1IkWi8USKTSmRyeUKVPKf38IO5HT5PNV3l8ZsCISgYUpaxRsjNKl+NTq9XqGo6iu1XU61IAsjAzhAYgBJcSScKZ7N5gtxMR8CBcXAsACCAQCsFwuBsMFziX85vqdJTVhsdgczENbg8/t2FpDESisQSyVS6SyuQKRXDMC9vJTFm6fQGQzSfDGZom01m80Wy1W6022wD5oORxOS3ODEui1u90ez1ezHeUqJ3ygcF3UBBg6GxWEVxKQR0UCLEKjxdciR9aUKgpKlaQZZkGHZTlEP5ZCUPMVVRUkcVExQxV+WTP9PlVDAoHVFNFRlIA)